src/cloudai/_core/test_scenario_parser.py

```diff
@@ -136,8 +139,29 @@ def _parse_data(self, data: Dict[str, Any]) -> TestScenario:
         total_weight = sum(tr.weight for tr in ts_model.tests)
         normalized_weight = 0 if total_weight == 0 else 100 / total_weight
 
+        pre_test, post_test = None, None
+        if ts_model.pre_test:
+            pre_test = self.hook_mapping.get(ts_model.pre_test)
+            if pre_test is None:
+                msg = (
+                    f"Pre-test hook '{ts_model.pre_test}' not found in hook mapping. "
+                    "Ensure that a proper hook directory is set under the working directory."
```

Comment:
Let's also print the file name and directory where we expect files to be.

Reply:
Updated.